src/codecoverage/bot/code_coverage_bot/zero_coverage.py

```diff
@@ -85,7 +86,61 @@ def get_fileinfo(self, filenames):
 
         return res
 
-    def generate(self, artifacts, hgrev, gitrev, out_dir='.'):
+    def generate_zero_coverage_some_platform_only(self, reports, hgrev, gitrev, out_dir='.'):
+        '''
+        :param reports: dictionary with key platform name, and value is report generated
+                       using ZeroCov.generate for that platform
+        :param hgrev: hg revision, must be same for all reports
+        :param gitrev: git revision, must be same for all reports
+        :param out_dir: directory to output the json result
+        '''
+
+        platforms = ['linux', 'windows']
+
+        git_revision = set([reports[platform]['github_revision'] for platform in platforms])
+        hg_revision = set([reports[platform]['hg_revision'] for platform in platforms])
+
+        assert len(git_revision) == 1 and len(hg_revision) == 1, 'Git revision and mercurial revision must be the same'
+
+        # mapping file_name -> set of platforms uncovered
+        uncovered_map = collections.defaultdict(lambda: set())
+        platform_files = collections.defaultdict(lambda: set())
+        all_files = {}
+
+        for platform in platforms:
+            for file_info in reports[platform]['files']:
+                uncovered_map[file_info['name']].add(platform)
```

Comment:
You are now adding files to uncovered_map even if they are not uncovered?

Reply:
sorry I was sleepy that moment, and wrongly understand what you meant by not needing the uncovered attribute. I will revert this part back